src/util.jl

```diff
@@ -19,3 +19,12 @@ function perpendicular_vector(vec::SVector{3})
     @inbounds perpind2 = vec[ind1]
     perp = @SVector [ifelse(i == ind1, perpind1, ifelse(i == ind2, perpind2, zero(T))) for i = 1 : 3]
 end
+
+"""
+    angle_difference(a, b)
+
+Compute the difference between two angles in such a way that:
+* for angles `a` and `b` (in radians), and `c = angle_difference(a, b)`, `cos(a) ≈ cos(b + c)` and `sin(a) ≈ sin(b + c)`;
+* the result is in the interval [-π, π).
```

Comment:
This is a useful utility (It's like deg2rad, I think I've written it several times before in various code bases :-) ).  I found this documentation a little confusing though.  How about something like:

"Compute the angle difference `a-b` wrapped into the interval [-π, π) by subtracting or adding a multiple of 2π.  This is the smallest number `c` such that `a` and `b+c` represent the same point on the circle, and likewise `b` and `a-c`."

Reply:
Hmm.  Smallest in the sense of `abs(c)` of course... could no doubt still be confusing.

Reply:
Done.

Reply:
Ah nice, you fixed up my awkward wording regarding the absolute value.